### PR TITLE
sensors: Add Angst+Pfister FXC-MDLx5-series driver 

### PR DESCRIPTION
**Pull request by jeppenodgaard**
Add driver for Angst+Pfister O2 sensors FCX-MLD25 & FCX-MLD95 and maybe
more. Tested with FCX-MLD25.

Supports get O2 value, get status, and power management.

Note that in suspended power mode heating output is at 20 %, thus
probably not suited for a battery powered device.

### CODE REVIEW

File: drivers/sensor/fcx_mldx5/fcx_mldx5.c [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+		return -EIO;
+	} else if (data->frame[FCX_MLDX5_STX_INDEX] != FCX_MLDX5_STX) {
+		LOG_WRN("No STX");
+		return -EIO;
+	} else if (strncmp(command, command_received, FCX_MLDX5_CMD_LEN) != 0) {
+		LOG_WRN("Expected command %s not %.*s", command, FCX_MLDX5_CMD_LEN,
+			command_received);
+		return -EIO;
+	} else if (data->frame[FCX_MLDX5_ETX_INDEX(data->frame_len)] != FCX_MLDX5_ETX) {
+		LOG_WRN("No ETX");
```
Comment (MaureenHelm) [inline drivers/sensor/fcx_mldx5/fcx_mldx5.c:169]: Is there a reason not to use `LOG_ERR` for all of these error cases?

Comment (jeppenodgaard) [reply drivers/sensor/fcx_mldx5/fcx_mldx5.c:169]: No really. Will fix!

Comment (jeppenodgaard) [reply drivers/sensor/fcx_mldx5/fcx_mldx5.c]: > Is there a reason not to use `LOG_ERR` for all of these error cases?

Fixed. Also pipeline passes now.

Comment (jeppenodgaard) [reply drivers/sensor/fcx_mldx5/fcx_mldx5.c]: Ping @MaureenHelm :)

**Review by MaureenHelm — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**